### DISCUSSION TITLE
Lock screen function

### DISCUSSION THREAD

**Discussion by JRosello** [KlipperScreen/KlipperScreen#178] [Ideas]
Can be created a button to lock the touch function of the screen?
Now I just installed and putted all together, very happy but in one of my first prints (5 hour print) with the screen attached, I switched off the backlight to low the heat and consumption and my mistake was assuming that turns off completely the screen so when I grabbed it and pressed without knowing the Emergency Stop button on the interface.

I don't know if it's doable or not, but as this issue happened to me maybe someone else can it happen too

**Comment by russiancatfood** [comment]
I second this. We take printers to shows and the number of times we have to both children and fully gown adults from messing with machines is alarming.

**Comment by alfrix** [comment]
this already has 15 votes, makes sense, i'll code it.

**Reply by sorted01** [reply]
Looking forward to that :D I was already thinking about printing a cover...

**Comment by alfrix** [comment]
done in 9a6aeaa43eebd0015d6bcb0dc66f1b6affcfd126

**Comment by jterranella** [comment]
Is it possible to have a variable that allows you to boot with the screen locked? If someone power cycled the printer, the password lock would be disabled unless it starts locked.
EX: Variable under [Main] boot_locked=true (or false)

 @alfrix

**Reply by jterranella** [reply]
Also, is there a way we can lock the screen during an active print? The place where the power button usually is when idle is the E-stop when printing.

**Reply by alfrix** [reply]
There is no extra functions at the moment, I'll take a look at it at some point, for now PRs are welcomed

**Comment by rati0nal** [comment]
How does the lock screen get initiated? I've added it to my klipperscreen.conf but it never appears to activate automatically.

**Reply by rati0nal** [reply]
Ok, the locking part required an update to Klipperscreen (accessible through the power menu for anyone reading) but I would love an option in config to toggle the following:

- Lock when printing (possibly with variable X minutes after print start)
- Lock on startup
- Toggle in the klipperscreen UI to toggle automatic locking

My use case is my printer sits in my garage and near a thoroughfare, it's easily knocked, the lock mechanism is handy to prevent this.
